### PR TITLE
Feature/fallback fonts

### LINKED ISSUE #5055: Include nerd font glyphs in Zed Mono
### Check for existing issues

- [X] Completed

### Describe the feature

I kinda like the font Zed is shipping with, Zed Mono, but it's currently not completely fit for also using in the builtin terminal, as I make heavy use of extra glyphs (like in [Nerd Fonts](https://www.nerdfonts.com)) both in my prompt and various applications. It would be nice if Zed Mono could become a “Nerd Font”!

### If applicable, add mockups / screenshots to help present your vision of the feature

_No response_

### LINKED ISSUE #5180: Multiple Fonts/Fallback Fonts
### Check for existing issues

- [X] Completed

### Is your feature request related to a problem?

I really love using Nerd Fonts or Powerline symbols in code to make my simple CLIs feel a little more immersive. I have problems with specific patched Nerd Fonts on MacOS, so I use a font with fallbacks in terminals and other places. The specific fonts I use in these are:
```
"Symbols Nerd Font Mono",
"Apple Color Emoji",
```
And this solves my issue in places like Wezterm where there is support for a font with multiple fallbacks.

> PS: The font Victor Mono's Nerd Font, either when custom patched or downloaded from the prepatched set, isn't compatible with MacOS, and that is why I use Symbols Nerd Font Mono.

### Describe the solution you'd like

I would like to see a font with fallback implementation come to Zed. Maybe just a way to pass in an array to `buffer_font_family` that uses the first font as the main font like normal. But then any after that would stand in for missing characters that don't exist in the main font.

### If applicable, add mockups / screenshots to help present your vision of the feature

In Wezterm, with my font with fallback configuration set, I get this when running my alias to `exa -A --icons` (exa is a modern replacement to LS with better colors and icon support).
image
In Zed however, I cannot set a font with fallback, so Zed replaces the characters that don't exist in Victor Mono, my main font, with small unknown icons to show it cannot find that glyph.
image
My proposed solution would make my config look like:
```json
{
    "buffer_font_size": 13,
    "buffer_font_family": ["Victor Mono", "Symbols Nerd Font Mono", "Apple Color Emoji"],
    "autosave": "on_focus_change",
    "vim_mode": true
}
```
And it would make the output In Zed and Wezterm match.

### PR DESCRIPTION
**Pull request by mikayla-maki**
Supersedes https://github.com/zed-industries/zed/pull/12090

fixes #5180
fixes #5055

See original PR for an example of the feature at work.

This PR changes the settings interface to be backwards compatible, and adds the `ui_font_fallbacks`, `buffer_font_fallbacks`, and `terminal.font_fallbacks` settings.

Release Notes:

- Added support for font fallbacks via three new settings: `ui_font_fallbacks`, `buffer_font_fallbacks`, and `terminal.font_fallbacks`.(#5180, #5055).

### CODE REVIEW

File: crates/settings/src/settings_file.rs [hunk 1]
```diff
@@ -15,12 +15,14 @@ pub fn test_settings() -> String {
     .unwrap();
     util::merge_non_null_json_value_into(
         serde_json::json!({
-            "ui_font_family": "Courier",
+            "ui_font_family": ["Courier"],
```
Comment (JunkuiZhang) [inline crates/settings/src/settings_file.rs:18]:
```suggestion
            "ui_font_family": "Courier",
```

I guess this fails the tests.

File: crates/settings/src/settings_file.rs [hunk 2]
```diff
@@ -15,12 +15,14 @@ pub fn test_settings() -> String {
...
             "ui_font_features": {},
             "ui_font_size": 14,
-            "buffer_font_family": "Courier",
+            "ui_font_fallback": [],
+            "buffer_font_family": ["Courier"],
```
Comment (JunkuiZhang) [inline crates/settings/src/settings_file.rs:22]: Same here.

File: crates/settings/src/settings_file.rs (hunk 1 above)
Comment (mikayla-maki) [reply crates/settings/src/settings_file.rs]: ugh, I can't believe I missed that. Thanks!

**Comment by AurevoirXavier** [comment]
Request to update the documentation website.

**Pull request merged**